### PR TITLE
Add Dataset Label Checking

### PR DESCRIPTION
**Pull request by ntalluri**

### CODE REVIEW

**Comment by ntalluri** [comment]
I'm not sure how to efficiently and correctly test this change. Should I test every potential way a user could make the dataset label?

File: test/test_config.py [hunk 1]
```diff
@@ -102,3 +102,6 @@ def test_config_container_registry(self):
         test_config["container_registry"]["owner"] = ""
         config.init_global(test_config)
         assert (config.config.container_prefix == config.DEFAULT_CONTAINER_PREFIX)
+
+    def test_dataset_label(self):
```
Comment (agitter) [inline test/test_config.py:106]: We don't need to test everything. Test coverage should make sure that ~2 valid labels pass and 2-3 invalid labels raise ValueErrors. Example invalid labels can contain `-`, another special character like `$`, and a Unicode symbol like `✉`.

We'll need to think about the most efficient way to test these. Do we need to write separate .yml files for each, or can we use a `sed` like operation to create temp copies of the `config.yml` file with modified dataset labels and then parse those?

**Review by agitter — COMMENTED** [review]
Testing ideas

File: test/test_config.py (hunk 1 above)
Comment (ntalluri) [reply test/test_config.py:106]: I tried something similar to the functions above in the file. It seems like I can just overwrite the temp_config and initialize the config to trigger the error.

**Review by agitter — COMMENTED** [review]
The testing approach you came up with is very clean. These are great.

It is a little unusual that `_` would be allowed as a dataset label, but let's leave that behavior in order to keep the regex simple.

We should still update `config.yml` to document the label name requirements. Besides that, I'll do some local testing and it will be ready.

**Comment by ntalluri** [comment]
Is ready to review

**Pull request merged**